restorers/dataloader/base/base_dataloader.py

```diff
@@ -28,69 +39,145 @@ def fetch_dataset(self, val_split: float, visualize_on_wandb: bool):
     def sanity_tests(self):
         raise NotImplementedError(f"{self.__class__.__name__ }.sanity_tests")
 
-    def read_image(self, image_path: str):
+    def read_image(self, image_path: str) -> tf.Tensor:
+        """
+        Function to read an image from the file path.
+
+        Args:
+            image_path (`str`): The image file path.
+
+        Returns:
+            tf.Tensor image in channel-last format.
+        """
         image = tf.io.read_file(image_path)
-        image = tf.image.decode_png(image, channels=3)
+        image = tf.io.decode_image(image, channels=3, expand_animations=False)
         image = tf.cast(image, dtype=tf.float32) / self.normalization_factor
         return image
 
-    def random_crop(self, input_image, gt_image):
-        input_image_shape = tf.shape(input_image)[:2]
-        crop_width = tf.random.uniform(
-            shape=(), maxval=input_image_shape[1] - self.image_size + 1, dtype=tf.int32
+    def random_crop(
+        self, input_image: tf.Tensor, enhanced_image: tf.Tensor
+    ) -> Tuple[tf.Tensor]:
+        """
+        Function to apply random cropping.
+
+        Args:
+            input_image (`tf.Tensor`): Low light image.
+            enhanced_image (`tf.Tensor`): Enhanced image.
+
+        Returns:
+            A tuple of random cropped image.
+        """
+        # Check whether the image size is smaller than the original image
+        image_size = tf.minimum(self.image_size, tf.shape(input_image)[0])
+
+        # Concatenate the low light and enhanced image
+        concatenated_image = tf.concat(
+            [input_image[None, ...], enhanced_image[None, ...]],
```

Comment:
Hi @ariG23498, as per https://github.com/soumik12345/mirnetv2/issues/17 we decided upon the removal of slice operations to make our data loading faster. I think we could avoid the slicing here entirely if we concatenate using the last axis instead of the first one. This is just a minor suggestion, feel free to ignore it.

Reply:
+1